src/bytes.rs

```diff
@@ -1488,7 +1485,7 @@ impl Borrow<[u8]> for BytesMut {
 
 impl fmt::Write for BytesMut {
     fn write_str(&mut self, s: &str) -> fmt::Result {
-        BufMut::put(self, s);
+        self.extend_from_slice(s.as_bytes());
```

Comment:
My opinion has been that you don't want your buffer to implicitly grow. This is due to `BufMut` implementations having an explicit capacity.

It could be that this should be changed... but there will be implications w/ the `BufMut` impl that would need to be thought through.